physical/postgresql/postgresql_test.go

```diff
@@ -260,13 +260,23 @@ func testPostgresSQLLockRenewal(t *testing.T, ha physical.HABackend) {
 	// Cancel attempt after lock ttl + 1s so as not to block unit tests forever
 	stopCh := make(chan struct{})
 	timeout := time.Duration(lock.ttlSeconds)*time.Second + lock.retryInterval + time.Second
-	time.AfterFunc(timeout, func() {
+
+	var leaderCh2 <-chan struct{}
+	newlockch := make(chan struct{})
+	go func() {
+		leaderCh2, err = newLock.Lock(stopCh)
+		close(newlockch)
+	}()
+
+	select {
+	case <-time.After(timeout):
 		t.Logf("giving up on lock attempt after %v", timeout)
 		close(stopCh)
-	})
+	case <-newlockch:
+		// pass through
+	}
 
 	// Attempt to lock should work
```

Comment:
Can we move this into the goroutine right after we lock?

Reply:
+1 - yeah, this confused me

Reply:
`t.Fatal` can't be used in a goroutine, so this is probably the right place for the error check.

Reply:
Sure thing, that makes sense. Done in db7641948

Reply:
Ah - today I learned:

> A test ends when its Test function returns or calls any of the methods FailNow, Fatal, Fatalf, SkipNow, Skip, or Skipf. Those methods, as well as the Parallel method, must be called only from the goroutine running the Test function. 

- https://golang.org/pkg/testing/#T

Reverted in 868288e39, thanks @briankassouf  !